### PR TITLE
GH workflow: introduce shellcheck job

### PR DESCRIPTION
**Pull request by msehnout**

### CODE REVIEW

**Comment by msehnout** [comment]
Is this something we want? Check the output, it is a fairly long list of errors/warinings for our shell scripts.

**Comment by dvdhrm** [comment]
I am generally in favor of this. Not sure how popular `shellcheck` is, but running linters on bash scripts seems like a highly useful CI tool.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -91,3 +91,10 @@ jobs:
 
       - name: Send coverage to codecov.io
         run: bash <(curl -s https://codecov.io/bash)
+  shellcheck:
+    name: Shellcheck
```
Comment (ondrejbudai) [inline .github/workflows/tests.yml:95]: The job name doesn't start with emoji. This violates our policies. :stuck_out_tongue_winking_eye:

**Review by ondrejbudai — COMMENTED** [review]
I love linters. ShellCheck seems to be pretty popular and well-maintained, I vote in favor of using it.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (gicmo) [reply .github/workflows/tests.yml:95]: :joy_cat:

**Comment by gicmo** [comment]
FWIF, I am all for it!

File: .github/workflows/tests.yml (hunk 1 above)
Comment (msehnout) [reply .github/workflows/tests.yml:95]: The koji sign is not an emoji! 麹 seems to be "malt" in English according to Google Translate :D

Comment (msehnout) [reply .github/workflows/tests.yml]: But I fixed it anyway...

Comment (ondrejbudai) [reply .github/workflows/tests.yml]: Love it! :shell: :shell: :shell:

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -91,3 +91,12 @@ jobs:
 
       - name: Send coverage to codecov.io
         run: bash <(curl -s https://codecov.io/bash)
+  shellcheck:
+    name: "🐚 Shellcheck"
+    runs-on: ubuntu-latest
```
Comment (larskarlitski) [inline .github/workflows/tests.yml:96]: I think @major will approve.

**Review by larskarlitski — APPROVED** [review]
This is great, thanks! All looks correct to me.

Is the commit that enables the GitHub action the last one? (It's hard to see in GitHub's time-based commit view.)

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -91,3 +91,14 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run ShellCheck
+      uses: ludeeus/action-shellcheck@master
```
Comment (gicmo) [inline .github/workflows/tests.yml:100]: Maybe pin this down to a specific tag?

Comment (msehnout) [reply .github/workflows/tests.yml:100]: Yes, that is a good idea. Thanks :). I'll use the same as you did in osbuild-koji.

**Comment by msehnout** [comment]
> This is great, thanks! All looks correct to me.
> 
> Is the commit that enables the GitHub action the last one? (It's hard to see in GitHub's time-based commit view.)

It wasn't, but now it is and it also uses the same well defined version of the shellcheck action as recommended by @gicmo . :)

**Review by gicmo — APPROVED** [review]
:shell: LGTM, Thanks!

File: schutzbot/mockbuild.sh [hunk 4]
```diff
@@ -88,27 +88,27 @@ fi
...
 # Copy the current build to the latest directory.
-mkdir -p $REPO_DIR_LATEST
-cp -arv ${REPO_DIR}/ ${REPO_DIR_LATEST}/
+mkdir -p "$REPO_DIR_LATEST"
+cp -arv "${REPO_DIR}"/ "${REPO_DIR_LATEST}"/
 
 # Remove the previous latest build for this branch.
 # Don't fail if the path is missing.
-s3cmd --recursive rm s3://${REPO_BUCKET}/${JOB_NAME}/latest/${ID}${VERSION_ID//./}_${ARCH} || true
+s3cmd --recursive rm s3://${REPO_BUCKET}/"${JOB_NAME}"/latest/"${ID}""${VERSION_ID//./}"_"${ARCH}" || true
```
Comment (ondrejbudai) [inline schutzbot/mockbuild.sh:111]:
```suggestion
s3cmd --recursive rm "s3://${REPO_BUCKET}/${JOB_NAME}/latest/${ID}${VERSION_ID//./}_${ARCH}" || true
```

Can we just quote the entire thing? Or is that not possible? FTR there are several cases of this in the PR.

**Review by ondrejbudai — DISMISSED** [review]
I'm not sure if I like the quoting changes (apparently created automatically by shellcheck) but it can be fixed later.

File: schutzbot/mockbuild.sh (hunk 4 above)
Comment (msehnout) [reply schutzbot/mockbuild.sh]: I think there should be no problem with it, but if the CI is happy with the current version I'd leave it as is and continue to improve our bash codebase in later PRs.

Anyway I changed at least this occurrence.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (major) [reply .github/workflows/tests.yml:96]: 🥰

**Review by ondrejbudai — DISMISSED** [review]

**Comment by ondrejbudai** [comment]
@msehnout can you rebase? #995 caused a conflict. :slightly_frowning_face:

**Review by ondrejbudai — APPROVED** [review]

**Comment by ondrejbudai** [comment]
Oh, I thought the conflict is resolved and it's apparently not. May I ask for one more force-push? :D

**Review by teg — APPROVED** [review]
Nice stuff!

**Review by ondrejbudai — APPROVED** [review]
CI fails because of #997. Otherwise, this is pure awesomeness.

**Pull request merged**